src/modules/ctc/src/ctc.cc

```diff
@@ -13,10 +11,534 @@
+{
+    return block_list_.size();
+}
+
+void GraphBuilder::AddSection(std::vector<types::Block> section, types::BlockDirection direction)
+{
+    for (size_t i = 0; i < section.size(); ++i)
+    {
+        graph_.AddEdge(section[i].block, section[i + 1].block, section[i + 1].length);
+    }
+    if (direction == types::BLOCKDIRECTION_BIDIRECTIONAL)
+    {
+        for (size_t i = section.size() - 1; i > 0; --i)
+        {
+            graph_.AddEdge(section[i].block, section[i - 1].block, section[i - 1].length);
+        }
+    }
+}
+
+Graph<types::BlockId, types::Meters> GraphBuilder::GetGraph(void)
+{
+    return graph_;
+}
+
+/*------------------------------ Ctc Implementation ------------------------------*/
+void Ctc::SetTrackLayout(void)
+{
+    CsvParser                 parser(schedule_file_path_);
+    BlockBuilder              bb(parser.GetRecords(), Module::MODULE_CTC);
+    std::vector<types::Block> blocks = bb.GetBlocks();
+    SetBlocks(blocks);
+    SetStations(blocks_);
+    SetGraphLayout(blocks_);
+}
 
-#endif // TRAINS_SRC_MODULES_CTC_INC_CTC_CC
\ No newline at end of file
+void Ctc::SetStations(std::vector<types::Block> &blocks)
+{
+    for (types::Block block : blocks)
+    {
+        if (block.has_station)
+        {
+            ctc::Station station(block.station_name, block.block, block.total_time_to_station);
+            stations_.push_back(station);
+        }
+    }
+}
+
+void Ctc::SetGraphLayout(std::vector<types::Block> &blocks)
+{
+    GraphBuilder gb(blocks);
+    gb.BuildGreenLine();
+    graph_ = gb.GetGraph();
+}
+
+void Ctc::SetGraphManually(Graph<types::BlockId, types::Meters> graph)
+{
+    graph_ = graph;
+}
+
+std::vector<types::BlockId> Ctc::FindRoute(const types::BlockId start, const types::BlockId end)
+{
+    return graph_.Dijkstra(start, end);
+}
+
+void Ctc::AssignAuthority(const std::vector<types::BlockId> &route, types::TrainId train_id)
+{
+    std::queue<types::BlockId> authority(std::deque<types::BlockId>(route.begin(), route.end()));
+
+    for (ctc::Train &train : train_schedules_)
+    {
+        if (train.train_id == train_id)
+        {
+            train.authority = authority;
+        }
+    }
+}
+
+
+/* Method: LoadSchedule()
+ * 1. Create CsvParser Object CsvParser(const std::filesystem::path &file_path) to get list of records
+ * 2. Create BlockBuilder Object to parse record information into blocks and populate ctc.blocks_ and ctc.train_schedules_
+ * 3. Upon receiving train schedule, tell train_model and train_controller to create according objects
+ * 4. Calculate authority and suggested speed for each train in train_schedules_
+ * 5. Calculate departure time for each train in ctc.train_schedule_
+ * 6. Set up timer for each train; when timer is up and match with depature time, get top block in authority queue and set suggested speed to block's speed limit
+ */
+void Ctc::LoadSchedule(void)
+{
+
+}
+
+types::Error Ctc::OpenFileExplorer(std::string &file)
```

Comment:
This should not be part of the CTC class.  This should be spun off to its own module in `common`